src/abstractdatatable/join.jl

```diff
@@ -2,18 +2,15 @@
 ## Join / merge
 ##
 
-# Like similar, but returns a nullable array
+# Like similar, but returns a array that can have nulls and is initialized with nulls
 similar_nullable{T}(dv::AbstractArray{T}, dims::Union{Int, Tuple{Vararg{Int}}}) =
-    NullableArray{T}(dims)
+    (v = Vector{Union{T, Null}}(dims); fill!(v, null); return v)
```

Comment:
Try simplifying to `similar(dv, Union{T, Null})` as suggested by @nalimilan 
 in https://github.com/JuliaData/DataTables.jl/pull/83#discussion_r135572418